HLTrigger/Timer/plugins/FastTimerService.cc

```diff
@@ -1766,13 +1776,25 @@ FastTimerService::Measurement& FastTimerService::ThreadGuard::thread() {
 }
 
 void FastTimerService::on_scheduler_entry(bool worker) {
-  if (guard_.register_thread(overhead_)) {
+  // The AtomicResources passed to register_thread are used to account the resources
+  // used or freed by the thread after the last active measurement and before leaving the TBB pool.
+  if (guard_.register_thread(job_summary_.idle)) {
     // initialise the measurement point for a thread that has newly joined the TBB pool
     thread().measure();
+  } else {
+    // An existing thread is re-joining the TBB pool.
+    // Note: unsure whether the resources used outside of the TBB pool should be
+    //   - not accounted:       thread().measure()
+    //   - considered as idle:  thread().measure_and_accumulate(job_summary_.idle)
+    //   - considered as other: thread().measure_and_accumulate(job_summary_.overhead)
+    thread().measure_and_accumulate(job_summary_.overhead);
```

Comment:
Other is probably the best guess, a thread isn't going to leave the primary pool unless it has actual work to steal from another pool.